components/project-loader/project-loader.jsx

```diff
@@ -1,51 +1,53 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Service from '../../js/service.js';
 import ProjectList from '../project-list/project-list.jsx';
 import pageSettings from '../../js/app-page-settings';
 import env from "../../config/env.generated.json";
 
 const PROJECT_BATCH_SIZE = env.PROJECT_BATCH_SIZE;
 
-export default React.createClass({
-  getInitialState() {
-    return {
+class ProjectLoader extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
       loadingData: false,
       nextBatchIndex: 1,
       entries: [],
       moreEntriesToFetch: false,
       totalMatched: 0
     };
-  },
+  }
+
   componentDidMount() {
     if (pageSettings.shouldRestore) {
       // restore state back to what is stored in pageSettings
       this.setState(pageSettings.currentList);
     } else {
       this.fetchData();
     }
-  },
+  }
+
   componentWillReceiveProps(nextProps) {
     // Reset state before fetching data for the new params.
     // We want to keep existingPromise on record for fetchData
     // to handle it accoringly.
 
-    this.setState(this.getInitialState(), () => {
```

Comment:
we don't need to revise code here if we keep getInitialState responsible for "what the initial state should look like". Always good to keep code from repeating itself, especially with hardcoded values.